src/CoreBundle/Bridge/Chameleon/Module/NavigationTree/NavigationTree.php

```diff
@@ -0,0 +1,740 @@
+        if ($startNodeId === $this->rootNodeId) {
+            $level = 1; //children of root are portals
+        }
+
+        $node = new \TdbCmsTree();
+        $node->SetLanguage(\TdbCmsUser::GetActiveUser()->GetCurrentEditLanguageID());
+        $node->Load($startNodeId);
+
+        $childrenArray = [];
+        $children = $node->GetChildren(true);
+        while ($child = $children->Next()) {
+            array_push($childrenArray, $this->createTreeDataModel($child, $level));
+        }
+        return $childrenArray;
+    }
+
+    private function createTreeDataModel(\TdbCmsTree $node, $level): BackendTreeNodeDataModel
+    {
+        $treeNodeDataModel = $this->backendTreeNodeFactory->createTreeNodeDataModelFromTreeRecord($node);
+
+        $treeNodeDataModel->setName($this->checkNameTranslation($treeNodeDataModel->getName(), $node));
+        $this->setTypeAndAttributes($treeNodeDataModel, $node);
+
+        // $level 0 == RootNode, 1 = Portal, >2 = Folder or Page
+        if ($level < 2) {
+            $liAttr = $treeNodeDataModel->getLiAttr();
+            $liAttr = array_merge($liAttr, ['class' => 'no-checkbox']);
+            $treeNodeDataModel->setLiAttr($liAttr);
+            $treeNodeDataModel->setOpened(true);
+        }
+
+        ++$level;
+        $children = $node->GetChildren(true);
+        while ($child = $children->Next()) {
+            $childTreeNodeDataModel = $this->createTreeDataModel($child, $level);
+            $treeNodeDataModel->addChildren($childTreeNodeDataModel);
+        }
+
+        return $treeNodeDataModel;
+    }
+
+    protected function checkNameTranslation(string $name, \TdbCmsTree $node): string
+    {
+        $cmsUser = \TCMSUser::GetActiveUser();
+        $editLanguage = $cmsUser->GetCurrentEditLanguageObject();
+        $node->SetLanguage($editLanguage->id);
+
+        if ('' === $name) {
+            $name = \TGlobal::OutHTML($this->translator->trans('chameleon_system_core.text.unnamed_record'));
+        }
+
+        if (true === CMS_TRANSLATION_FIELD_BASED_EMPTY_TRANSLATION_FALLBACK_TO_BASE_LANGUAGE) {
+            $cmsConfig = \TCMSConfig::GetInstance();
+            $defaultLanguage = $cmsConfig->GetFieldTranslationBaseLanguage();
+            if (null !== $defaultLanguage && $defaultLanguage->id !== $editLanguage->id) {
+                if ('' === $node->sqlData['name__'.$editLanguage->fieldIso6391]) {
+                    $name .= ' <span class="bg-danger px-1"><i class="fas fa-language" title="'.$this->translator->trans('chameleon_system_core.cms_module_table_editor.not_translated').'"></i></span>';
+                }
+            }
+        }
+
+        return $name;
+    }
+
+    private function setTypeAndAttributes(BackendTreeNodeDataModel $treeNodeDataModel, \TdbCmsTree $node): void
+    {
+        $treeNodeDataModel->setType($this->setInitialType($node));
+
+        $liAttr = $treeNodeDataModel->getLiAttr();
+        $aAttr = [];
+        $liClass = '';
```

Comment:
I named the variables that way because that's their name in the jstree json configuration (li_attr and a_attr). Therefore I find my naming more comprehensible.

Reply:
Variablen sind umbenannt.